Update 'Course' implementation

griffine opened the pull request:
Updated versions of the Conceptual Schema, Logical Schema, and createTables.sql have been uploaded to fix issue #7 (fixing the implementation for table 'Course'). 

The updated Conceptual Schema (GeeksConceptualSchema.pdf) has attribute 'Credits' added to the Course entity. The updated Logical Schema (GEEKSLogicalSchema.doc) has attribute 'Credits' added to relation Course, as well as an update in the data type for attribute 'Name'; VARCHAR(8) was changed to VARCHAR(10) to accommodate for courses offered by multiple departments. The corresponding changes were made to the Course table in the createTables.sql.

Commit:
Add Updated Conceptual Schema

Add a PDF Conceptual Schema based off of the M2 Conceptual Schema from Gradebook. Added the attribute 'Credits' to the 'Course' entity.

Commit:
Add Updated Logical Schema

Add 'GEEKSLogicalSchema.doc'. This file is an updated logical schema that corresponds to the 'GEEKSConceptualSchema.pdf', and it is originally based on the M2 Logical Schema from Gradebook.

Commit:
Update createTables.sql

Apply changes to Course table to correspond to updated schema. Add 'Credits' attribute, and change value of attribute 'Number' from VARCHAR(8) to VARCHAR(10).

kozlowskik reviewed (commented):
Changes overall look good, the only concern that I can see happening is in the chance that for example writing and English both list the same course that is a writing intensive course `WRT/ENG102W` because this would add up to be 11 characters long and the character limit expressed it 10

griffine commented:
Thanks for pointing that out - I had forgotten that some of the writing courses have the 'W' tagged onto the end of the course number.

Commit:
Update GEEKSLogicalSchema.doc

Change the data type for 'Number' in Course from VARCHAR(10) to VARCHAR(11) to account for longer course numbers such as 'WRT/ENG101W'.

Commit:
Update createTables.sql

Change the data type for 'Number' in the Course table from VARCHAR(10) to VARCHAR(11) to account for course numbers that are longer (i.e. 'WRT/ENG103W').
